server/app/controllers/admin/AdminProgramStatusesController.java

```diff
@@ -86,9 +88,27 @@ public Result createOrUpdate(Http.Request request, long programId)
       // form values.
       return ok(statusesView.render(request, program, Optional.of(form)));
     }
+    // Because we need to look directly at the value sent (or absent) in the form data
+    // itself to determine if a checkbox is checked, we need to calculate this here
+    // before passing the value into createOrEditStatusFromFormData. If the checkbox
+    // is not checked, no value is sent. If it is checked, "on" is sent for the value.
+    Optional<String> maybeSetDefault =
+        Optional.ofNullable(form.rawData().get(ProgramStatusesForm.DEFAULT_CHECKBOX_NAME));
+    final boolean setDefault;
+    if (!maybeSetDefault.isPresent() || maybeSetDefault.get().isBlank()) {
```

Comment:
At this point, we want to see what the state of the checkbox is.  `form.get().getDefaultStatus()` will still return `false` when we've got the checkbox set, since we haven't actually updated the status to make it default yet.  This is determining if we should be setting the status default to true when we call `createOrEditStatusFromFormData`. 

That said, I've got a bit of tunnel vision on this now.  I can't remember exactly why `defaultStatus` on the form isn't set to the state of the checkbox on submission.  Something about not being able to tell a checkbox input to set it's value on form submission I think.

Reply:
@dkatzz Does that look right to you?